routes/api.php

```diff
@@ -23,7 +24,13 @@
 
 Route::controller(RegisterController::class)->group(function(){
     Route::post('register', 'register');
-    Route::post('login', 'login');
+    Route::post('login/admin', 'loginAdmin'); 
+    Route::post('login/customer', 'loginCustomer');
+});
+
+Route::controller(UsersController::class)->group(function(){
```

Comment:
Do this for admin
```
Route::middleware('auth.admin:sanctum')->group(function () {
   Route::post(...)
});
```
For customers
```
Route::middleware('auth.customer:sanctum')->group(function () {
   Route::post(...)
});
```

There will also be setups to make this work

Reply:
@clmnshn28 

### 1. Define Custom Guards in `auth.php`
In your `config/auth.php` file, you can define custom guards for `admin` and `customer`:

```php
'guards' => [
    'admin' => [
        'driver' => 'sanctum',
        'provider' => 'admins',
    ],

    'customer' => [
        'driver' => 'sanctum',
        'provider' => 'customers',
    ],

    // Default web and API guards...
    'web' => [
        'driver' => 'session',
        'provider' => 'users',
    ],

    'api' => [
        'driver' => 'sanctum',
        'provider' => 'users',
    ],
],

'providers' => [
    'admins' => [
        'driver' => 'eloquent',
        'model' => App\Models\Admin::class,
    ],

    'customers' => [
        'driver' => 'eloquent',
        'model' => App\Models\Customer::class,
    ],

    // Default user provider...
    'users' => [
        'driver' => 'eloquent',
        'model' => App\Models\User::class,
    ],
],
```

### 2. Create Middleware for Admin and Customer
You can create middleware to enforce authentication using these custom guards.

For example, create `AdminAuthMiddleware` and `CustomerAuthMiddleware`:

```bash
php artisan make:middleware AdminAuthMiddleware
php artisan make:middleware CustomerAuthMiddleware
```

In `AdminAuthMiddleware`, check for the `admin` guard:

```php
public function handle($request, Closure $next)
{
    if (!Auth::guard('admin')->check()) {
        return response()->json(['error' => 'Unauthorized'], 401);
    }

    return $next($request);
}
```

In `CustomerAuthMiddleware`, check for the `customer` guard:

```php
public function handle($request, Closure $next)
{
    if (!Auth::guard('customer')->check()) {
        return response()->json(['error' => 'Unauthorized'], 401);
    }

    return $next($request);
}
```

### 3. Register Middleware in `Kernel.php`
Register these middlewares in `app/Http/Kernel.php`:

```php
protected $routeMiddleware = [
    'auth.admin' => \App\Http\Middleware\AdminAuthMiddleware::class,
    'auth.customer' => \App\Http\Middleware\CustomerAuthMiddleware::class,
    // Other middlewares...
];
```

### 4. Apply Middleware to Routes
You can now use these middlewares in your routes:

```php
Route::middleware('auth.admin')->group(function () {
    Route::get('/admin/dashboard', [AdminController::class, 'dashboard']);
    // Other admin routes...
});

Route::middleware('auth.customer')->group(function () {
    Route::get('/customer/dashboard', [CustomerController::class, 'dashboard']);
    // Other customer routes...
});
```

### 5. Protect Resources with Specific Guards
For resource routes, you can also specify the middleware:

```php
Route::middleware('auth.admin')->resource('admin/products', ProductController::class);
Route::middleware('auth.customer')->resource('customer/orders', OrderController::class);
```

This setup ensures that only authenticated admins can access the admin routes and only authenticated customers can access the customer routes, using Sanctum for token-based authentication.

Reply:
When you're going to send request from frontend you will add `Authorization` in your axios header.
```
{
  'Authorization': `Bearer ${token}`
}
```